x/incentives/keeper/store.go

```diff
@@ -143,6 +143,33 @@ func (k Keeper) GetAllGroupsGauges(ctx sdk.Context) ([]types.Gauge, error) {
 	return gauges, nil
 }
 
+// GetAllGroupsWithGauge iterates through all groups, sequentially pulls the gauges from each, and returns the groups with their associated gauge.
+func (k Keeper) GetAllGroupsWithGauge(ctx sdk.Context) ([]types.GroupsWithGauge, error) {
```

Comment:
Let's also cover this with a simple test plase

Reply:
done https://github.com/osmosis-labs/osmosis/pull/6594/commits/aa8c9e4bff03161ef00e495e1254106a0d77e54b